client/src/fluid/Fluid.res

```diff
@@ -1000,6 +1008,35 @@ let caretTargetForNextNonWhitespaceToken = (~pos, tokens: tokenInfos): option<CT
   getNextWS(tokens)
 }
 
+@ocaml.doc("Gets the caret target for a pos within a list of tokens. Ignores
+  whitespace tokens, and favors tokens 'to the left' of the pos, in cases where two
+  tokens surround the position.")
+let caretTargetForCurrentNonWhitespaceToken = (~pos, tokens: tokenInfos): option<CT.t> =>
+  List.fold(
```

Comment:
I'm not 100% sure, but it feels like this could be a `List.find` instead of a `fold`

Reply:
I couldn't figure out how :) Open to suggestions!